### DISCUSSION TITLE
Deserialisation of custom values

### DISCUSSION THREAD

**Discussion by shershen0** [pdvrieze/xmlutil#134] [Q&A]
I need to parse this kind of xml: 

```
<?xml version="1.0" encoding="UTF-8"?>
<project
        xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
        xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">

    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    </properties>

</project> 
```

but I cannot figure out how to parse `properties/project.build.sourceEncoding` element 

I would like to see something like this in my Data Class, but this still doesn't work 

```
@Serializable
@SerialName("properties")
data class Properties(
    @XmlElement(true) val properties: List<String>? = null
)
```

currently I get this error: which I suppose means it cannot find any field where it could be written: 
```
nl.adaptivity.xmlutil.serialization.UnknownXmlFieldException: Could not find a field for name (data.Project) {http://maven.apache.org/POM/4.0.0}project/{http://maven.apache.org/POM/4.0.0}properties (Element)
  candidates: 

```

Would appreciate your help!

**Comment by shershen0** [comment]
also there could be xml like this: 

```
                   <tasks>
                           <delete dir="${basedir}/src/main/java" />
                           <mkdir dir="${basedir}/src/main/java" />
                           <copy todir="${basedir}/src/main/java">
                                 <fileset dir="${basedir}/../../../community/platform/annotations/src" />
                           </copy>
                   </tasks>
```

with completely random attributes and values and I have no clue right now how to write common code for all possible values

**Reply by pdvrieze** [reply]
I assume you are dealing with a "not-quite-xml" format that uses tag names dynamically. Have a look at: [examples/DYNAMIC_TAG_NAMES.md](https://github.com/pdvrieze/xmlutil/blob/master/examples/DYNAMIC_TAG_NAMES.md)

**Reply by pdvrieze** [reply]
Alternatively you can use DOM support (have a list of `Element`)
